### DISCUSSION TITLE
Could video streaming from the file server be possible?

### DISCUSSION THREAD

**Discussion by redtrillix** [rejetto/hfs#73] [Ideas]
As the title says, being able to stream videos without having to download them would be amazing. This already works with music and i like using that.

I have no experience with this sort of thing so i dont know the limitations and if its even worth it, just thought id chuck that idea out there.

**Comment by rejetto** [comment]
This is far from being clear, but it's the browser that decides to play content instead of downloading it.
This decision is normally taken solely based on the type of the content, and the type (also known as mime-type) is communicated by the server to the browser.
This information is configurable in HFS in admin/configuration, and you can see that by default it includes \*.mp3 .
All other extensions just don't "declare" their type, causing the browser to think it's just a bunch of bytes.
You may just have understood that adding `*.mp4` could do what you want,
but I suspect a simple `*` (catch all) is the best option for most users.

_TL;DR Can you  replace the mime type configuration with `*` and tell me if your experience is good?_
People may still want to download and not play. This should normally be possible by right clicking the link, even tho some people may miss this option.

Of course current situation is not ideal, but I didn't make any change because I'm not sure what's best.

Having an option like "enable streaming for videos" means
- i'm lying, i don't have the power to force this (specific) playing mechanism of the browser
- it doesn't work on all file formats
- it could conflict with the "mime type" configuration, because their scope is actually overlapping

These problems would lead to an inconsistent experience.

I think it would be best for next versions to go with the "*".
Opinions?

**Reply by redtrillix** [reply]
Oh cool, I wasn't aware that's what mime type did. I really appreciate it.

I've set mime type to `*` and so far, everything is downloading or showing as I want (e.g. executables downloading, mp4's playing in browser). I'll keep it as a wildcard and see what doesn't work out.
